### PR TITLE
Refactor to MirageJS

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation
Screen Shot 2020-07-14 at 4 57 15 PM
This is part of an on-going initiative to completely migrate all of `folio-org` projects from `@bigtest/mirage` to `miragejs`.

[`@bigtest/mirage`](https://github.com/bigtestjs/mirage/) was originally a prototype to make `mirage` available to any framework, but it has since been archived in favor of [`miragejs`](https://miragejs.com/). The documentation for `miragejs` is not only prodigious for its extensive API docs but also for its tutorials and guides. Furthermore, it receives regular bug-fixes and maintenance releases in thanks to its extremely large and active community.

Previous PR: [@folio-org/ui-local-kb-admin #123](https://github.com/folio-org/ui-local-kb-admin/pull/123)
_Linking the previous pull request so that subsequent ones can be found tagged below._

<details>
<summary>Overview of progress at the time of the creation of this PR (in chronological order):</summary>
<br>
✅ <code>ui-checkin</code>
<br>
✅ <code>ui-checkout</code>
<br>
☑️ <code>ui-circulation</code>
<br>
☑️ <code>ui-calendar</code>
<br>
✅ <code>ui-agreements</code>
<br>
☑️ <code>ui-app-template</code>
<br>
☑️ <code>ui-inventory</code>
<br>
☑️ <code>ui-data-export</code>
<br>
✅ <code>ui-finc-config</code>
<br>
☑️ <code>ui-licenses</code>
<br>
☑️ <code>ui-erm-comparisons</code>
<br>
☑️ <code>ui-myprofile</code>
<br>
☑️ <code>ui-local-kb-admin</code>
<br>
☑️ <code>ui-marccat</code>
</details>

## Approach
- [x] Cloned, installed, and then ran tests to double check that the tests are passing before refactoring.
- [x] Removed `@bigtest/mirage` and added `inflected` and `miragejs`.
- [x] ESlint & ran tests after the refactor to confirm they all pass.
- [x] ⚠️ Tests weren't setup for this project so there wasn't much to refactor except updating dependencies and some linting issues. I removed `husky` as well because it was pointless.

### CODE REVIEW

**Comment by minkimcello** [comment]
@zburke Reverted the `husky` removal. I shouldn't have removed it as it has nothing to do with mirage.

File: src/components/Search/Button/EditRecord.js [hunk 1]
```diff
@@ -24,11 +23,10 @@ const EditRecordButton = ({ ...props }) => {
     const { router, toggleFilterPane } = props;
     const id = findParam('id');
     toggleFilterPane();
-    // lockRecord(id);
+    lockRecord(id);
```
Comment (zburke) [inline src/components/Search/Button/EditRecord.js:26]: This changes the behavior of the edit button. Why is it included in this PR?

Comment (matteopascucci) [reply src/components/Search/Button/EditRecord.js:26]: I would also say to leave it commented for the moment

Comment (minkimcello) [reply src/components/Search/Button/EditRecord.js:26]: It might've been a lint thing. I'll undo it.

File: package.json [hunk 2]
```diff
@@ -30,34 +30,38 @@
...
     "immutable": "^3.7.5",
+    "inflected": "^2.0.4",
     "lodash": "^4.17.4",
+    "miragejs": "^0.1.40",
     "mocha": "^5.2.0",
     "moment": "^2.22.2",
     "qs": "^6.5.1",
+    "react": "^16.5.0",
+    "react-dom": "^16.5.0",
+    "react-intl": "^5.3.2",
```
Comment (zburke) [inline package.json:62]: We're only on v4.5 at present. Also, they only published `v4.0.0` less than six months ago, and `v5.0.0` less than a month ago and they're already on v5.4!!! So much churn; ugh.

Comment (zburke) [reply package.json:62]: (Sorry; just noticed this now; I know you've been through many versions of this PR. I'm not trying to kill you with 1000 small cuts, promise.)

Comment (minkimcello) [reply package.json]: > Sorry; just noticed this now; I know you've been through many versions of this PR. I'm not trying to kill you with 1000 small cuts, promise.

No worries 👌 

I believe i added those dependencies to remove some of the errors from `yarn install` but I went ahead and downgraded them to the same version as what's listed in peerDependency.

**Comment by minkimcello** [comment]
@zburke Do you know how I can get that SonarCloud check to pass?

File: src/components/Search/Button/EditRecord.js [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+  //   const payload = {
+  //     user,
+  //     id,
+  //     idLock,
+  //   };
+  //   dispatch(lockRecordAction(payload));
+  // };
   const handleClickEdit = () => {
     const { router, toggleFilterPane } = props;
     const id = findParam('id');
```
Comment (zburke) [inline src/components/Search/Button/EditRecord.js:24]: Ah, now I see why you uncommented `lockRecord`: Lint won't let you keep the function around if it's not called anywhere. 👌

**Comment by zburke** [comment]
Don't worry about the SonarCloud complaint. In this case its analysis doesn't hold up, and I don't think it blocks the merge.

**Pull request merged**